openconfig/ygot
# Add support for leaf getters.

```
* (M) ygen/*
   - Add support for leaf getters which enable the value of a leaf to be retrieved
     using Get<LeafName> methods. These getters return the Go zero value if a leaf
     is unset in the schema. Some caution must therefore be exercised such that
     applications that use them do not assume that a leaf is set simply because
     GetXXX returns a value.
   - A subsequent change will implement YANG default lookups such that GetXXX
     returns the YANG default for unset leaves.
 * (M) ygen/testdata/*
   - Mechanical change to make the receiver name for a generated Go struct
     consistent per style rules. We had a mix of s and t prior to this change.
 * (M) ygen/schema_tests/schema_test.go
   - Add tests against the generated code to ensure operation of leaf getters.
 * (M) generator/generator.go
   - Add support for the -generate_leaf_getters flag.
 * (M) exampleoc/*
   - Regenerate with leaf getters enabled.
```
